### PR TITLE
[WIP] basic event and rules handling

### PR DESCRIPTION
**Pull request by anandijain**
so currently im doing the most naive thing for the case of `S1 < 0.1` and converting it into a continuous event `S1 ~ 0.1`, just to see how far I get.

There's some big hacks here, for instance, any trigger more complicated than this will surely fail. As pointed out by Fredrik, I may need to come up with a smarter Callback type or something.

I'll need to make it so that ReactionSystem can get the events kwarg (which it might already, I just haven't tried).
I need to fix that states in the trigger are lacking the IV, I think I need to walk the tree and check if any of the args are in the states/species.
I'll want a real test that case 26 actually produces the same results as the test suite. 

Longer term, I'll probably want to do all that stuff with delays, discrete events (maybe), and the `useValuesFromTriggerTime` attribute. That'll surely be in another PR.

### CODE REVIEW

**Comment by anandijain** [comment]
also its annoying but juliaformatter did its thing and spaced everything differently

**Comment by anandijain** [comment]
@paulflang hope you're well, if you have a chance, you might want to take note that rules are soon to be finished and maybe review that they are handled correctly. 
My quick checks looks to me that I'm getting the same results as the suite for the three types of rules. 

Events are harder and I'm hoping to get them done by the end of the week

File: src/reactionsystem.jl [hunk 1]
```diff
@@ -3,19 +3,32 @@ function Catalyst.ReactionSystem(model::SBML.Model; kwargs...)  # Todo: requires
...
+    ODESystem(equations(sys);
+        name = nameof(sys),
+        defaults = ModelingToolkit.get_defaults(rs),
+        continuous_events = get_events(model, rs))
 end
 
 """ Check if conversion to ReactionSystem is possible """
 function checksupport(filename::String)
-    not_implemented = ["listOfRules", "listOfConstraints", "listOfEvents"]
+    not_implemented = ["listOfRules", "listOfConstraints"]
```
Comment (paulflang) [inline src/reactionsystem.jl]: @anandijain I thought listOfRules are implemented now, but listOfEvents not.

Comment (anandijain) [reply src/reactionsystem.jl]: just fixed that

**Comment by anandijain** [comment]
I'm getting a bunch of failures from the new SBML which is why this PR is out of shape, hoping to resolve that today

File: src/reactionsystem.jl (hunk 1 above)
Comment (anandijain) [reply src/reactionsystem.jl]: oh i misread this. yeah events are still wip. thinking on it again ill probably split this pr up and do rules first

**Review by paulflang — APPROVED** [review]
Looks good to me. Thanks @anandijain !

**Pull request closed without merging**